### PR TITLE
fix(thorvg): support rendering in draw events

### PR DESCRIPTION
**Pull request by kisvegabor**
### Description of the feature or fix

see #6378

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by kisvegabor** [comment]
@FASTSHIFT The tests are passing for me locally. Does it pass on your end?

File: demos/vector_graphic/lv_demo_vector_graphic.c [hunk 1]
```diff
@@ -240,11 +240,19 @@ static void draw_vector(lv_layer_t * layer)
...
     lv_draw_buf_t * draw_buf = lv_canvas_get_draw_buf(obj);
     lv_draw_buf_destroy(draw_buf);
 }
 
+static void event_cb(lv_event_t * e)
+{
+    lv_layer_t * layer = lv_event_get_layer(e);
+
+    draw_vector(layer);
+
```
Comment (FASTSHIFT) [inline demos/vector_graphic/lv_demo_vector_graphic.c]: Remove extra line breaks.

File: src/libs/thorvg/tvgSwRenderer.cpp [hunk 2]
```diff
@@ -443,11 +443,7 @@ bool SwRenderer::target(pixel_t* data, uint32_t stride, uint32_t w, uint32_t h,
 
 bool SwRenderer::preRender()
 {
-#if LV_USE_DRAW_VG_LITE && LV_USE_VG_LITE_THORVG
 	return true;
```
Comment (FASTSHIFT) [inline src/libs/thorvg/tvgSwRenderer.cpp]: Replace Tab with Space.

File: src/widgets/lottie/lv_lottie.c [hunk 3]
```diff
@@ -219,14 +219,13 @@ static void lottie_update(lv_lottie_t * lottie, int32_t v)
 
     lv_draw_buf_t * draw_buf = lv_canvas_get_draw_buf(obj);
     if(draw_buf) {
-#if LV_USE_DRAW_VG_LITE && LV_USE_VG_LITE_THORVG
         /**
          * Since the buffer clearing operation in the SwRenderer::preRender
          * function is removed when the VG-Lite simulator is enabled, the canvas
          * buffer must be manually cleared here.
          */
```
Comment (FASTSHIFT) [inline src/widgets/lottie/lv_lottie.c]: Remove unnecessary comments

**Comment by FASTSHIFT** [comment]
> @FASTSHIFT The tests are passing for me locally. Does it pass on your end?

It may be related to the PR: https://github.com/lvgl/lvgl/pull/6402 that was just merged. Have you tried rebasing?

**Comment by kisvegabor** [comment]
[Failing](https://github.com/lvgl/lvgl/actions/runs/9647837028/job/26609076735?pr=6406) after rebase too :slightly_frowning_face:

**Comment by FASTSHIFT** [comment]
> [Failing](https://github.com/lvgl/lvgl/actions/runs/9647837028/job/26609076735?pr=6406) after rebase too 🙁

I reproduced the problem locally and uploaded new reference images.

**Review by kisvegabor — COMMENTED** [review]
It's interesting that my test locally doesn't fail with the new reference image. Anyway, it seems working.

File: src/widgets/lottie/lv_lottie.c (hunk 3 above)
Comment (kisvegabor) [reply src/widgets/lottie/lv_lottie.c]: fixed

File: src/libs/thorvg/tvgSwRenderer.cpp (hunk 2 above)
Comment (kisvegabor) [reply src/libs/thorvg/tvgSwRenderer.cpp]: Fixed

File: demos/vector_graphic/lv_demo_vector_graphic.c (hunk 1 above)
Comment (kisvegabor) [reply demos/vector_graphic/lv_demo_vector_graphic.c]: Fixed

File: demos/vector_graphic/lv_demo_vector_graphic.c [hunk 4]
```diff
@@ -240,11 +240,19 @@ static void draw_vector(lv_layer_t * layer)
 
 static void delete_event_cb(lv_event_t * e)
 {
+    LV_UNUSED(e);
```
Comment (liamHowatt) [inline demos/vector_graphic/lv_demo_vector_graphic.c]: It is used. Please remove

**Review by liamHowatt — COMMENTED** [review]
Otherwise LGTM!

File: demos/vector_graphic/lv_demo_vector_graphic.c (hunk 4 above)
Comment (kisvegabor) [reply demos/vector_graphic/lv_demo_vector_graphic.c]: Thanks, fixed!

**Review by liamHowatt — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**